### PR TITLE
[ECAM] Support CONF 3 Landing

### LINKED ISSUE #803: [BUG] GPWS LDG Flap 3
**Mod Version**

DEV Version, and the newest MSFS version

**Describe the bug**

GPWS LDG Flap 3 shows wrong info -> When its turned off, it displays boxed "OFF" info, which i think it shouldnt because this buttons default state should be OFF. Also, when its turned off, the ECAM shows that its turned on.

**To Reproduce**

1.  Click on the LDG FLAP 3 button on the Overhead panel -> GPWS

**Expected behavior**

The button should be OFF by default, and when its off it shouldnt display an ECAM message + it should display only the "ON" light on the button when its turned on

**Actual behavior**

The button is in the ON position by default, and when its off it shows an ECAM message and the OFF light on the button

**References**

image

Discord username (if different from GitHub): LorDawid#6403

### PR DESCRIPTION
**Pull request by tracernz**
I have split the ECAM half off #2889 in this PR.
Fixes #803 (the remainder of)

## Summary of Changes

The final item on the ECAM landing memo is flaps. Whether this item refers to flaps 3 or full depends on the GPWS LDG FLAP 3 switch on the overhead panel. The logic to enable GPWS warnings at flaps 3 or full depending on the switch already exists.

## Screenshots (if necessary)

Before:
image

After:
image
image

## References

FCOM for the ECAM memo:
image

## Additional context

Discord username (if different from GitHub): tracernz

## Testing instructions

NOTE: This PR does not implement the MCDU part, so the target approach speed (magenta triangle on PFD speed tape) will still be that for flaps full. FMCG/MCDU work to come in #2889.
- GPWS warnings disabled while near ground with CONF 3 (and GPWS CONF 3 switch ON). This was already working before the PR.
- ECAM landing memo displaying correct flaps config and action.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by beheh** [comment]
The same condition that switches the LDG MEMO to FLAP should also add a green `GPWS FLAP 3` memo to the right column (based on the GPWS button). Order-/Priority-wise it's one of the lowest items in the list, before the AUTO BRKs and after the APU AVAIL/BLEED and LDG LT memo.

**Comment by tracernz** [comment]
> The same condition that switches the LDG MEMO to FLAP should also add a green `GPWS FLAP 3` memo to the right column (based on the GPWS button). Order-/Priority-wise it's one of the lowest items in the list, before the AUTO BRKs and after the APU AVAIL/BLEED and LDG LT memo.

Thanks @beheh. The message was already there, but looking at the wrong L var, so I have fixed that in 982b729. One thing I note is that `L:A32NX_GPWS_FLAPS3` is fetched quite a few times and as I understand it simvars are a little slow. Would it be worth getting it once per update and putting it in a member variable for use in the various places? I guess these kinds of problems go away when the EWD ECAM gets re-written for React if we have a SimVar watcher?

**Comment by beheh** [comment]
>One thing I note is that L:A32NX_GPWS_FLAPS3 is fetched quite a few times and as I understand it simvars are a little slow. Would it be worth getting it once per update and putting it in a member variable for use in the various places?

A helper for this already exists in the UpperECAM. You can call `this.getCachedSimVar`.

> I guess these kinds of problems go away when the EWD ECAM gets re-written for React if we have a SimVar watcher?

Yes, although it will be structured quite differently as the message activation logic will live in WASM, and the EWD itself will have to only look at a few SimVars to determine which message is shown.

**Comment by tracernz** [comment]
> A helper for this already exists in the UpperECAM. You can call `this.getCachedSimVar`.

Didn't notice I was already using it in https://github.com/flybywiresim/a32nx/pull/3140/commits/982b729f7a6dfa12ccfeda9dfd075ac8dffe2a7b >.<. Replaced the rest of the accesses.
> 
> Yes, although it will be structured quite differently as the message activation logic will live in WASM, and the EWD itself will have to only look at a few SimVars to determine which message is shown.

Brilliant.

**Review by beheh — APPROVED** [review]

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@
 1. [SOUND] Reworked engine spool up and startup sounds - @hotshotp (Boris)
 1. [SOUND] Reworked pack sounds - @hotshotp (Boris)
 1. [SOUND] Reworked touchdown sounds - @hotshotp (Boris)
+1. [ECAM] Add CONF/FLAPS 3 memos when GPWS LDG FLAPS 3 enabled - @tracernz (Mike)
```
Comment (MisterChocker) [inline .github/CHANGELOG.md:45]:
```suggestion
1. [ECAM] Added CONF/FLAPS 3 memos when GPWS LDG FLAPS 3 enabled - @tracernz (Mike)
```

**Review by MisterChocker — CHANGES_REQUESTED** [review]
tempus

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 24/01/2021
Version of the sim: 1.12.13.0
PR Tested: #3140 
Tier of testing: 1
Changes to observe: ECAM CONF 3 indication

Testing technique: Tryout

Steps you've made to test the Pull Request:

1. Spawn on VABB Approach for RWY27.
2. Turned on GPWS FLAP 3, checked ECAM (Screenshot 1)
3. Set Flaps to 3. (Screenshot 2)
4. Continued approach. Tried turning off GPWS FLAP 3, ECAM mode reverted to FLAP FULL config.

Media:
image

image

image

Issues (If Any): None

Overall Rating: Good
Conclusions of the testing: Since it's an ECAM visual addition, not tested in detail, looks good to go. Can't wait for the other half of the PR (MCDU) to be implemented! 💯

**Comment by stescouse** [comment]
Category: QA Tester
Name: Discord User stescouse#7271
Date of testing: 24.1.2021
Version of the sim: 1.12.13.0
PR Tested: #3140
Tier of testing: 1
Changes to observe: [ECAM] Support CONF 3 Landing.

Testing technique: TryOut
Steps you've made to test the Pull Request:

1. Spawned C&D at GMME
2. Flew to LFPG.
3. On APPR engaged GPWS FLAP 3
flap 3 - 2

4. GPWS FLAP 3 displayed correctly in ECAM.

flap 3 - 1

5. On Final APPR engaged FLAPS 3 and displayed correctly in ECAM with no config warning.

flap 3 - 3

6  After landing messages cleared correctly in ECAM

flap 3 - 4


Flight notes (If required: Full flight from GMME to LFPG.

Issues (If Any): None;


Overall Rating: Good
Conclusions of the testing: Performed as set out in PR and as stated on MDCU approach speed for PFD is still FLAPS FULL.

**Review by MisterChocker — APPROVED** [review]

**Pull request merged**